connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -941,10 +1018,20 @@ private void addRequest(Callable<Void> action, Callback<Void> callback) {
     private void addRequest(long delayMs, Callable<Void> action, Callback<Void> callback) {
         HerderRequest req = new HerderRequest(time.milliseconds() + delayMs, action, callback);
         requests.add(req);
-        if (requests.peek() == req)
+        if (peekWithoutException() == req)
             member.wakeup();
     }
 
+    private HerderRequest peekWithoutException() {
+        try {
+            return requests.isEmpty() ? null : requests.first();
```

Comment:
More compact I agree. But they are not equivalent. 

The need for the above comes from the fact that you can't "peek" the head of the skip list without throwing an exception if you peek on an empty set. And you don't want to throw an exception every time you peek on an empty set. 

Thus, the above tries its best not to throw an exception, by checking first whether the set is empty. Each of `isEmpty` and `first` is atomic but not their combination. So there's a pretty slim chance that you'll execute `first` on an empty set, although you just checked whether it was empty or not. But that's the best you can do to avoid throwing an exception, and should be good enough here.